internal/pkg/build/assemblers/assembler_sif.go

```diff
@@ -215,9 +217,79 @@ func (a *SIFAssembler) Assemble(b *types.Bundle, path string) error {
 		return fmt.Errorf("while creating SIF: %v", err)
 	}
 
+	labels := make(map[string]string, 1)
+
+	// Get the old image labels first
+	if b.RunSection("labels") && len(b.Recipe.ImageData.Labels) > 0 {
+		for key, value := range b.Recipe.ImageData.Labels {
+			labels[key] = value
+		}
+	}
+
+	sylog.Infof("Inserting Metadata Labels...")
+
+	// load the container to add the metadata
+	fimg, err := sif.LoadContainer(path, false)
+	if err != nil {
+		return fmt.Errorf("failed to load sif container file: %s", err)
+	}
+	defer fimg.UnloadContainer()
+
+	descr, err := getDescr(&fimg)
+	if err != nil {
+		return fmt.Errorf("no primary partition found: %s", err)
+	}
+	groupid := descr[0].Groupid
+
+	metadata.GetImageInfoLabels(labels, &fimg, b)
+
+	// make new map into json
+	text, err := json.MarshalIndent(labels, "", "    ")
+	if err != nil {
+		return err
+	}
+
+	// Add the metadata
+	err = sifAddMetadata(&fimg, groupid, uint32(0), text)
+	if err != nil {
+		return fmt.Errorf("failed adding metadata block to SIF container file: %s", err)
+	}
+
+	return nil
+}
+
+func sifAddMetadata(fimg *sif.FileImage, groupid, link uint32, data []byte) error {
+	// data we need to create a signature descriptor
+	siginput := sif.DescriptorInput{
+		Datatype: sif.DataLabels,
+		Groupid:  groupid,
+		Link:     link,
+		Fname:    "image-metadata",
+		Data:     data,
+	}
+	siginput.Size = int64(binary.Size(siginput.Data))
+
+	// add new signature data object to SIF file
+	err := fimg.AddObject(siginput)
+	if err != nil {
+		return err
+	}
+
 	return nil
 }
 
+func getDescr(fimg *sif.FileImage) ([]*sif.Descriptor, error) {
```

Comment:
Is this helper really needed? It's just wrapping another function

Reply:
Yes, I think it is needed. I will put in a common package with a better func name.